debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerStreamingChangeEventSource.java

```diff
@@ -447,4 +458,34 @@ private Lsn getToLsn(SqlServerConnection connection, String databaseName, TxLogP
 
         return connection.getNthTransactionLsnFromLast(databaseName, fromLsn, maxTransactionsPerIteration);
     }
+
+    @Override
+    public void commitOffset(Map<String, ?> offset) {
+        Lsn commitLsn = Lsn.valueOf((String) offset.get("commit_lsn"));
+        synchronized (streamingExecutionContexts) {
+            for (Map.Entry<SqlServerPartition, SqlServerStreamingExecutionContext> entry : streamingExecutionContexts.entrySet()) {
```

Comment:
This looks a bit odd that we need to iterate over the contexts for all partitions while committing an offset that belongs to a specific partition.

Probably the current Debezium API doesn't account for this but here we need to know some key (e.g. the source partition) that the offset being committed belongs to. Otherwise, iterating over a hundred partitions when committing the offset of each will likely hurt performance.

Reply:
It would be more optimal but we'd still have to iterate the list. We don't need to address this problem immediately, let's focus on the rest of the concerns first.

Reply:
Jira ticket: refactor this in order to avoid iteration by using the partition to which the offset being committed belongs (need some research on Kafka Connect API first).